### DISCUSSION TITLE
Does --config-path or CCACHE_CONFIGPATH have priority?

### DISCUSSION THREAD

**Discussion by dyadav7** [ccache/ccache#986] [Questions & answers] [answered]
### Higher preference  between --config-path and CCACHE_CONFIGPATH ###

```
--config-path PATH
Let the command line options operate on configuration file PATH instead of the default. Using this option has the 
same effect as setting the environment variable CCACHE_CONFIGPATH temporarily.
```
If both --config-path and env CCACHE_CONFIGPATH are used then which is actually considered by ccache?
This is not very clear from the docs.

**Comment by jrosdahl** [comment, marked as answer]
Hi @dyadav7,

> If both --config-path and env CCACHE_CONFIGPATH are used then which is actually considered by ccache ?

`--config-path`.

> This is not very clear from the docs.

I thought the documentation is quite clear – _"Using this option has the same effect as setting the environment variable CCACHE_CONFIGPATH temporarily"_ means that using `--config-path` is equivalent to setting `CCACHE_CONFIGPATH` immediately before executing ccache. And if you set `CCACHE_CONFIGPATH` immediately before executing ccache it will naturally override any previous setting of `CCACHE_CONFIGPATH`. That said, I'm open to improvements of the documentation if you have any good suggestions.

**Comment by dyadav7** [comment]
I guess this is better:

```
--config-path PATH
Let the command line options operate on configuration file PATH. This option has 
higher precedence over the environment variable CCACHE_CONFIGPATH.
```

The section "Configuration" should also mention  --config-path :
e.g.
```
Location of the primary configuration file

The location of the primary (cache-specific) configuration is determined like this:

1. If option --config-path is used, use that path.   <== NOTE: This needs to be added
2. If CCACHE_CONFIGPATH is set, use that path.

```

**Reply by jrosdahl** [reply]
(Note: You created a new answer to your own question instead of commenting my answer. Consider writing replies in the _Write a reply_ area instead of the _Suggest an answer_ area next time.)

> I guess this is better:
> 
> ```
> --config-path PATH
> Let the command line options operate on configuration file PATH. This option has 
> higher precedence over the environment variable CCACHE_CONFIGPATH.
> ```

That version removes the distinction that usage of the option is optional (previously it said "instead of the default"). I like the current formulation (saying that usage of the option has the same effect as setting CCACHE_CONFIGPATH) because the option then doesn't have to be documented in other places as well.

> The section "Configuration" should also mention --config-path : e.g.
> 
> ```
> Location of the primary configuration file
> 
> The location of the primary (cache-specific) configuration is determined like this:
> 
> 1. If option --config-path is used, use that path.   <== NOTE: This needs to be added
> 2. If CCACHE_CONFIGPATH is set, use that path.
> ```

I don't think that's a good idea since the `--config-path` options is a special case that is not involved in normal ccache operation, only when running ccache without a compiler.

**Reply by jrosdahl** [reply]
I made a small improvement in c83c3ccdd3d5.

**Reply by dyadav7** [reply]
The small improvement is better. This disccussion can be closed.
